client/ui.c

```diff
@@ -99,7 +99,7 @@ void PrintAndLogEx(logLevel_t level, char *fmt, ...) {
 		}
 		PrintAndLog(buffer2);
 	} else {
-		snprintf(buffer2, sizeof(buffer2), "%s%s", prefix, buffer);
+		snprintf(buffer2, sizeof(buffer2), "%s%.2028s", prefix, buffer);
```

Comment:
What if MAX_PRINT_BUFFER is redefined?

Reply:
The problem is that both buffer2 and buffer are MAX_PRINT_BUFFER sized.
The concatenation of buffer and prefix can overflow buffer2.

Reply:
I do understand the issue but I am wondering whether there is another solution. If someone changes MAX_PRINT_BUFFER to e.g. 1024, the warning would appear again.

Can we change the size of either buffer or buffer2?

Or would something like
```
snprintf(buffer2, sizeof(buffer2), "%s%." MAX_PRINT_BUFFER-20 "s", prefix, buffer);
```
work?

Reply:
I agree !
The good syntax is:
`snprintf(buffer2, sizeof(buffer2), "%s%.*s", prefix, MAX_PRINT_BUFFER - 20, buffer);`